pkg/cloud/provider/provider.go

```diff
@@ -149,12 +149,20 @@ func ShareTenancyCosts(p models.Provider) bool {
 // NewProvider looks at the nodespec or provider metadata server to decide which provider to instantiate.
 func NewProvider(cache clustercache.ClusterCache, apiKey string, config *config.ConfigFileManager) (models.Provider, error) {
 	nodes := cache.GetAllNodes()
-	if len(nodes) == 0 {
-		log.Infof("Could not locate any nodes for cluster.") // valid in ETL readonly mode
-		return &CustomProvider{
-			Clientset: cache,
-			Config:    NewProviderConfig(config, "default.json"),
-		}, nil
+	var numRetries int
+	for len(nodes) == 0 {
+		log.Infof("No nodes found for cluster, will retry...")
+		time.Sleep(time.Second)
+		nodes = cache.GetAllNodes()
+
+		numRetries++
+		if numRetries == 10 {
+			log.Infof("Could not locate any nodes for cluster.") // valid in ETL readonly mode
```

Comment:
Note, if I can't figure this out, and this code is required. You'll only want to implement the retry for the following condition: 

```
if nodes == 0 && !env.IsETLReadOnlyMode()
```

Reply:
This is because k8s-less installs use a default provider and expect this scenario.

Reply:
There is also an `github.com/opencost/opencost/core/pkg/util/retry` utility for this behavior if interested. https://github.com/opencost/opencost/blob/develop/core/pkg/util/retry/retry.go

Reply:
Updated! I appreciate the feedback.